assembly/src/ast/mod.rs

```diff
@@ -177,11 +193,20 @@ impl ProgramAst {
         // asserts below are OK because we enforce limits on the number of procedure and the
         // number of body instructions in relevant parsers
 
-        assert!(self.local_procs.len() <= u16::MAX as usize, "too many local procs");
+        assert!(self.imports.len() <= MAX_LOCAL_PROCS, "too many imports");
+        target.write_u16(self.imports.len() as u16);
+        // We don't need to serialized the keys, since they are the last name in the path
+        let mut values: Vec<LibraryPath> = Vec::<LibraryPath>::new();
+        for (_, path) in self.imports.iter() {
+            values.push(path.clone());
+        }
```

Comment:
minor nit: any reason not to use use `self.imports.values()` here?

Reply:
Yes: I'm still learning the standard library. :-)

Fixed.